File: aipmb_manage/lib/models/user_tag.dart
class UserTag {
  final String userName;
  final List<TagItem> tags;
  final String generatedAt;
  final String modelUsed;

  UserTag({
    required this.userName,
    required this.tags,
    required this.generatedAt,
    required this.modelUsed,
  });

  factory UserTag.fromJson(Map<String, dynamic> json) {
    return UserTag(
      userName: json['user_name'] ?? '',
      tags: (json['tags'] as List<dynamic>?)
              ?.map((t) => TagItem.fromJson(t))
              .toList() ??
          [],
      generatedAt: json['generated_at'] ?? '',
      modelUsed: json['model_used'] ?? '',
    );
  }
}

class TagItem {
  final String name;
  final String reasoning;

  TagItem({required this.name, required this.reasoning});

  factory TagItem.fromJson(Map<String, dynamic> json) {
    return TagItem(
      name: json['name'] ?? '',
      reasoning: json['reasoning'] ?? '',
    );
  }
}

File: pmb/ai_manage/models/user_tag.py
"""用户标签数据模型"""
from dataclasses import dataclass, field
from datetime import datetime


@dataclass
class UserTag:
    user_name: str
    tags: list[dict] = field(default_factory=list)  # [{"name": "...", "reasoning": "..."}]
    generated_at: str = field(default_factory=lambda: datetime.now().isoformat())
    model_used: str = ""

    def to_dict(self) -> dict:
        return {
            "user_name": self.user_name,
            "tags": self.tags,
            "generated_at": self.generated_at,
            "model_used": self.model_used,
        }

    @classmethod
    def from_dict(cls, d: dict) -> "UserTag":
        return cls(
            user_name=d.get("user_name", ""),
            tags=d.get("tags", []),
            generated_at=d.get("generated_at", ""),
            model_used=d.get("model_used", ""),
        )